src/Fixer/FunctionNotation/VoidReturnFixer.php

```diff
@@ -0,0 +1,251 @@
+                }
+
+                continue;
+            }
+
+            if ($this->hasReturnAnnotation($tokens, $index)) {
+                continue;
+            }
+
+            $endIndex = $tokens->findBlockEnd(Tokens::BLOCK_TYPE_CURLY_BRACE, $startIndex);
+
+            if ($this->hasVoidReturn($tokens, $startIndex, $endIndex)) {
+                $this->fixFunctionDefinition($tokens, $startIndex);
+            }
+        }
+    }
+
+    /**
+     * Determine if there is a non-void return annotation in the function's PHPDoc comment.
+     *
+     * @param Tokens $tokens
+     * @param int    $index  The index of the function token
+     *
+     * @return bool
+     */
+    private function hasReturnAnnotation(Tokens $tokens, $index)
+    {
+        foreach ($this->findReturnAnnotations($tokens, $index) as $return) {
+            $types = $return->getTypes();
+            if (count($types) > 1 || 'void' !== $types[0]) {
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+    /**
+     * Determine if there is a void return annotation in the function's PHPDoc comment.
+     *
+     * @param Tokens $tokens
+     * @param int    $index  The index of the function token
+     *
+     * @return bool
+     */
+    private function hasVoidReturnAnnotation(Tokens $tokens, $index)
+    {
+        foreach ($this->findReturnAnnotations($tokens, $index) as $return) {
+            $types = $return->getTypes();
+            if (1 === count($types) && 'void' === $types[0]) {
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+    /**
+     * Determine the function already has a return type hint.
+     *
+     * @param Tokens $tokens
+     * @param int    $index  The index of the end of the function definition line, EG at { or ;
+     *
+     * @return bool
+     */
+    private function hasReturnTypeHint(Tokens $tokens, $index)
+    {
+        $endFuncIndex = $tokens->getPrevTokenOfKind($index, [')']);
+        $nextIndex = $tokens->getNextMeaningfulToken($endFuncIndex);
+
+        return $tokens[$nextIndex]->isGivenKind(CT::T_TYPE_COLON);
+    }
+
+    /**
+     * Determine if the function has a void return.
+     *
+     * @param Tokens $tokens
+     * @param int    $startIndex Start of function body
+     * @param int    $endIndex   End of function body
+     *
+     * @return bool
+     */
+    private function hasVoidReturn(Tokens $tokens, $startIndex, $endIndex)
+    {
+        for ($i = $startIndex; $i < $endIndex; ++$i) {
+            if ($tokens[$i]->isGivenKind(T_YIELD)) {
+                return false; // Do not apply fix as generators cannot return void.
+            }
+            if (!$tokens[$i]->isGivenKind(T_RETURN)) {
+                continue;
+            }
+
+            $nextToken = $tokens->getNextMeaningfulToken($i);
+            if (!$tokens[$nextToken]->equals(';')) {
+                return false; // Do not apply fix, non-empty return statement found.
```

Comment:
The comment is not relevant here IMO. You should move it to the main tokens iteration or remove it.